### DISCUSSION TITLE
enthusiactically waiting for webR to be more practical

### DISCUSSION THREAD

**Discussion by sbalci** [quarto-dev/quarto-cli#4778] [Show and Tell]
Hi,

With https://github.com/coatless/quarto-webr extension it will be easy to run webR :)

One point is that it is not practical to run a local html file without a localhost like server is enabled. webR is activated on github page or when rendered within Rstudio, but not active when I open the html file directly.

The location and use of `webr-worker.js` and `webr-serviceworker.js` should be explained more clearly.

I am enthusiactically waiting for webR to be more practical.

Serdar

**Comment by mcanouil** [comment]
This seems to be a discussion either on the webR repository or on the Quarto extension.m repository.

I don't believe this has anything to do with Quarto CLI itself at this point.

**Reply by sbalci** [reply]
sure just wanted to let people in the discussion to know new possibilities.

**Reply by mcanouil** [reply]
Well if that is your intent, I would suggest to rewrite your message that way, because right now your message is not about new possibilities brought by webR and the Quarto extension.

**Reply by sbalci** [reply]
thanks. better if an editor deletes this then, i couldn't. I will reopen the discussion with more information. sorry for this.

**Reply by mcanouil** [reply]
No apologies needed, nothing wrong. 😉
Simply edit your original post, no reasons to delete it in my opinion.

**Comment by coatless** [comment]
@sbalci Thanks for the feedback. As `quarto-webr` is an extension to Quarto, you probably want to place this post under the repository's issue tracker instead of the Quarto CLI one. 

<https://github.com/coatless/quarto-webr/issues>

That said, please note that opening the rendered webR-enabled Quarto HTML document directly is a limitation built-in to web browsers. We discuss a few options to overcome it here and provide further context here: 

<https://github.com/coatless/quarto-webr#directly-accessing-rendered-html>
